routemaster/client.rb

```diff
@@ -9,17 +11,22 @@
 
 module Routemaster
   class Client
-    
+
     def initialize(options = {})
-      @_url = _assert_valid_url(options[:url])
-      @_uuid = options[:uuid]
-      @_timeout = options.fetch(:timeout, 1)
-      @_verify_ssl = options.fetch(:verify_ssl, true)
+      @_options = options.tap do |o|
+        o[:timeout] ||= 1
+        o[:backend_type] ||= Backends::Synchronous
+      end
+
+      @_backend_type = @_options.fetch(:backend_type)
```

Comment:
You can avoid tapping before and just pass the default backend if they key is missing:
```
@_backend_type = @_options.fetch(:backend_type, Backends::Syncronous)
```
same goes for `@_timeout` which is actually in the previous implementation.